File: app/src/main/java/com/niu/protect/accessibility/auto/app/appinfo/AppInfo.java
package com.niu.protect.accessibility.auto.app.appinfo;

import android.content.Context;
import android.content.pm.ApplicationInfo;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.graphics.Bitmap;
import android.graphics.drawable.BitmapDrawable;
import android.graphics.drawable.Drawable;
import android.text.TextUtils;
public class AppInfo {
    static String appName = "";

    public static synchronized String getAppName(Context context) {
        String str;
        synchronized (AppInfo.class) {
            try {
                PackageManager packageManager = context.getPackageManager();
                if (TextUtils.isEmpty(appName)) {
                    appName = String.valueOf(packageManager.getApplicationLabel(context.getApplicationInfo()));
                }
            } catch (Exception e) {
                e.printStackTrace();
                appName = "3985学生端";
            }
            str = appName;
        }
        return str;
    }

    public static synchronized String getVersionName(Context context) {
        String str;
        synchronized (AppInfo.class) {
            try {
                PackageManager packageManager = context.getPackageManager();
                PackageInfo packageInfo = packageManager.getPackageInfo(context.getPackageName(), 0);
                str = packageInfo.versionName;
            } catch (Exception e) {
                e.printStackTrace();
                return null;
            }
        }
        return str;
    }

    public static synchronized int getVersionCode(Context context) {
        int i;
        synchronized (AppInfo.class) {
            try {
                PackageManager packageManager = context.getPackageManager();
                PackageInfo packageInfo = packageManager.getPackageInfo(context.getPackageName(), 0);
                i = packageInfo.versionCode;
            } catch (Exception e) {
                e.printStackTrace();
                return 0;
            }
        }
        return i;
    }

    public static synchronized String getPackageName(Context context) {
        String str;
        synchronized (AppInfo.class) {
            try {
                PackageManager packageManager = context.getPackageManager();
                PackageInfo packageInfo = packageManager.getPackageInfo(context.getPackageName(), 0);
                str = packageInfo.packageName;
            } catch (Exception e) {
                e.printStackTrace();
                return null;
            }
        }
        return str;
    }

    public static synchronized Bitmap getBitmap(Context context) {
        ApplicationInfo applicationInfo;
        Bitmap bm;
        synchronized (AppInfo.class) {
            PackageManager packageManager = null;
            try {
                packageManager = context.getApplicationContext().getPackageManager();
                applicationInfo = packageManager.getApplicationInfo(context.getPackageName(), 0);
            } catch (PackageManager.NameNotFoundException e) {
                applicationInfo = null;
            }
            Drawable d = packageManager.getApplicationIcon(applicationInfo);
            BitmapDrawable bd = (BitmapDrawable) d;
            bm = bd.getBitmap();
        }
        return bm;
    }
}

File: protect/src/main/java/com/niu/protect/lib/appinfo/AppInfo.kt
package com.niu.protect.lib.appinfo

import android.content.Context
import android.content.pm.ApplicationInfo
import android.content.pm.PackageManager
import android.graphics.Bitmap
import android.graphics.drawable.BitmapDrawable
import android.text.TextUtils
import com.niu.protect.core.Constants

object AppInfo {
    var appName = ""
    @JvmStatic
    @Synchronized
    fun getAppName(context: Context): String {
        var str: String
        synchronized(AppInfo::class.java) {
            try {
                val packageManager = context.packageManager
                if (TextUtils.isEmpty(appName)) {
                    appName = packageManager.getApplicationLabel(context.applicationInfo).toString()
                }
            } catch (e: Exception) {
                e.printStackTrace()
                appName = Constants.APP_NAME
            }
            str = appName
        }
        return str
    }

    @Synchronized
    fun getVersionName(context: Context): String? {
        var str: String
        synchronized(AppInfo::class.java) {
            str = try {
                val packageManager = context.packageManager
                val packageInfo = packageManager.getPackageInfo(context.packageName, 0)
                packageInfo.versionName
            } catch (e: Exception) {
                e.printStackTrace()
                return null
            }
        }
        return str
    }

    @Synchronized
    fun getVersionCode(context: Context): Int {
        var i: Int
        synchronized(AppInfo::class.java) {
            i = try {
                val packageManager = context.packageManager
                val packageInfo = packageManager.getPackageInfo(context.packageName, 0)
                packageInfo.versionCode
            } catch (e: Exception) {
                e.printStackTrace()
                return 0
            }
        }
        return i
    }

    @Synchronized
    fun getPackageName(context: Context): String? {
        var str: String
        synchronized(AppInfo::class.java) {
            str = try {
                val packageManager = context.packageManager
                val packageInfo = packageManager.getPackageInfo(context.packageName, 0)
                packageInfo.packageName
            } catch (e: Exception) {
                e.printStackTrace()
                return null
            }
        }
        return str
    }

    @Synchronized
    fun getBitmap(context: Context): Bitmap {
        var applicationInfo: ApplicationInfo?
        var bm: Bitmap
        synchronized(AppInfo::class.java) {
            var packageManager: PackageManager? = null
            try {
                packageManager = context.applicationContext.packageManager
                applicationInfo = packageManager.getApplicationInfo(context.packageName, 0)
            } catch (e: PackageManager.NameNotFoundException) {
                applicationInfo = null
            }
            val d = packageManager!!.getApplicationIcon(applicationInfo!!)
            val bd = d as BitmapDrawable
            bm = bd.bitmap
        }
        return bm
    }
}